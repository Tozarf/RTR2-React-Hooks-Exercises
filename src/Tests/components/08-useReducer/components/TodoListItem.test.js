import React from 'react'
import { shallow } from 'enzyme'
import "@testing-library/jest-dom"
import { TodoListItem } from '../../../../components/08-useReducer/components/TodoListItem'

import { demoTodos } from "../../../fixtures/demoTodos"


describe('testing <TodoListItem/>', () => {

    const handleDelete = jest.fn()
    const handleToggle = jest.fn()

    const [todo1]= demoTodos
    const wrapper = shallow(
        <TodoListItem todo={todo1} i={0} handleDelete={handleDelete} handleToggle={handleToggle}/>)
        
    test('should match snapshot', () => {

        expect(wrapper).toMatchSnapshot()   
    })
    test('should call handleDelete func', () => {
        

        wrapper.find("button").simulate("click")

        expect ( handleDelete ).toHaveBeenCalledWith(todo1.id)

    })
    test('should call handleToggle func', () => {
        
        wrapper.find("p").simulate("click")

        expect( handleToggle ).toBeCalledWith(todo1.id)

    })

    test('should show the text correctly', () => {
        
        const text = wrapper.find("p")
        expect(text.text().trim()).toEqual(`1. ${todo1.desc}`)


    })
    test('should have complete className if TODO.done=true', () => {
        
        todo1.done=true
        const wrapper = shallow(
            <TodoListItem todo={todo1}/>)
        
        expect(wrapper.find("p").hasClass("complete")).toBe(true)

    })

    

    
    
})
