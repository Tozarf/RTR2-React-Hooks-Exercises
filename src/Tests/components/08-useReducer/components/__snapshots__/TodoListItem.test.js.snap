// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <TodoListItem/> should match snapshot 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    . 
    Learn Mongo
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
